src/module/system/actions/base.ts

```diff
@@ -0,0 +1,43 @@
+import { ActionGlyph } from "@system/action-macros/types";
+import { Action, ActionUseOptions } from "@system/actions/types";
+
+interface BaseActionData {
+    description?: string;
+    glyph?: ActionGlyph;
+    name: string;
+    traits?: string[];
+}
+
+abstract class BaseAction implements Action {
+    readonly #description?: string;
+    readonly #glyph?: ActionGlyph;
```

Comment:
Storing a glyph is a little inline-brained. It should take an action cost, and a glyph derived from that (or not for exploration/downtime activities).